ext/compressors/iaa/iaacodec/include/IAACompressionCodecDeflate.h

```diff
@@ -0,0 +1,218 @@
+#pragma once
+
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include <unistd.h>
+#include <chrono>
+#include <sys/mman.h>
+#include <iostream>
+#include <atomic>
+#include <qpl/qpl.h>
+#include <vector>
+#include <map>
+#include <x86intrin.h>
+#include <memory>
+#include <thread>
+#include <wiredtiger.h>
+#include <wiredtiger_ext.h>
+namespace DB
+{
+
+class DeflateJobHWPool
+{
+public:
+    DeflateJobHWPool(WT_COMPRESSOR *compressor, WT_SESSION *session);
+    ~DeflateJobHWPool();
+    static DeflateJobHWPool & instance(WT_COMPRESSOR *compressor = nullptr, WT_SESSION *session = nullptr);
+    static constexpr auto jobPoolSize = 256;
+    static constexpr qpl_path_t PATH = qpl_path_hardware;
+    static qpl_job * jobPool[jobPoolSize];
+    static std::atomic_bool jobLock[jobPoolSize];
+    bool jobPoolEnabled;
+    std::atomic_uint64_t jobIndex;
+
+    inline bool jobPoolReady()
+    {
+        return jobPoolEnabled;
+    }
+    inline qpl_job * acquireJob(uint32_t * job_id)
+    {
+        if(jobPoolEnabled)
+        {
+            uint32_t retry = 0;
+            auto index = jobIndex.fetch_add(1) % jobPoolSize;
+            while (tryLockJob(index) == false)
+            {
+                retry++;
+                if (retry > jobPoolSize)
+                {
+                    return nullptr;
+                }
+                index = jobIndex.fetch_add(1) % jobPoolSize;
+            }
+            *job_id = jobPoolSize - index;
+            return jobPool[index];
+        }
+        else
+        {
+            return nullptr;
+        }
+    }
+    inline qpl_job * releaseJob(uint32_t job_id)
+    {
+        if(jobPoolEnabled)
+        {
+            uint32_t index = jobPoolSize - job_id;
```

Comment:
Should we have a protection in case `job_id > jobPoolSize`?

Reply:
OK
We will update it , Thanks!
        if(!jobPoolEnabled || job_id > jobPoolSize || job_id == 0)
            return nullptr;
        else
        {
            uint32_t index = jobPoolSize - job_id;
            ReleaseJobObjectGuard _(index);
            return jobPool[index];
        }

Reply:
Thanks for the update, I don't think we need the `else` statement here:
```suggestion
            if(!jobPoolEnabled ||  job_id == 0 || job_id > jobPoolSize)
                return nullptr;
            uint32_t index = jobPoolSize - job_id;
            ReleaseJobObjectGuard _(index);
            return jobPool[index];
```

Reply:
Yes, got it.